module/VuFind/src/VuFind/Auth/Manager.php

```diff
@@ -754,6 +773,25 @@ public function login($request)
                 throw new AuthException('authentication_error_technical', 0, $e);
             }
 
+            // Attempt catalog login so that any bad credentials are cleared before further processing
+            // (avoids e.g. multiple login attempts by account AJAX checks):
+            if ($this->ilsAuthenticator && $username = $user->getCatUsername()) {
+                try {
+                    $patron = $this->ils->patronLogin(
+                        $username,
+                        $this->ilsAuthenticator->getCatPasswordForUser($user)
+                    );
+                    if (empty($patron)) {
+                        // Problem logging in -- clear user credentials so they can be
+                        // prompted again; perhaps their password has changed in the
+                        // system!
```

Comment:
This might be problematic in combination with the `allowUserLogin = false` setting in config.ini. That might just be another argument for making the behavior configurable, but maybe we should revise the comment to reflect that possibility too...

Reply:
I changed the mechanism so that it's disabled with `allowUserLogin = false`.